src/components/CheckBoxList/check-box-item.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { Checkbox, ListItemIcon } from '@mui/material';
+import { SxProps } from '@mui/material';
+import OverflowableText from '../OverflowableText';
+
+export interface CheckBoxItemProps {
+    item: any;
+    checkBoxIconSx?: SxProps;
+    labelSx?: SxProps;
+    checked: boolean;
+    label: string;
+    onClick: (
+        event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+        item: any
+    ) => void;
+    secondaryAction?: React.ReactElement | undefined;
+    disabled: boolean | undefined;
+}
+
+const CheckBoxItem: FunctionComponent<CheckBoxItemProps> = ({
+    item,
+    checkBoxIconSx,
+    checked,
+    labelSx,
+    label,
+    onClick,
+    disabled = false,
+    secondaryAction,
+    ...props
+}) => {
+    return (
+        <>
+            <ListItemIcon sx={checkBoxIconSx}>
+                <Checkbox
+                    color={'primary'}
+                    edge="start"
+                    checked={checked}
+                    onClick={(event) => onClick(event, item)}
+                    disableRipple
+                    disabled={disabled}
+                    {...props}
+                />
+            </ListItemIcon>
+            <OverflowableText sx={labelSx} text={label} />
+            {secondaryAction && secondaryAction}
```

Comment:
```suggestion
            {secondaryAction}
```
I think it should work ?